Addons/testExoticQuadrature.hpp

```diff
@@ -0,0 +1,223 @@
+    bool passed = true;
+    double precision = 1e-10;
+    auto sinc = [freq](double x)->double {return (x == 0.0 ? 1.0 : sin(freq * x) / (freq * x));};
+    TasGrid::CustomTabulated ct = TasGrid::getExoticQuadrature(level, shift, sinc);
+    TasGrid::TasmanianSparseGrid sg;
+    sg.makeGlobalGrid(1, 1, 2*level-1, TasGrid::type_qptotal, std::move(ct));
+    std::vector<double> quad_points = sg.getPoints();
+    std::vector<double> quad_weights = sg.getQuadratureWeights();
+    double approx_integral = 0.0;
+    assert(quad_weights.size() == quad_points.size());
+    for (size_t i=0; i<quad_weights.size(); i++) {
+        approx_integral += f(quad_points[i]) * quad_weights[i];
+    }
+    if (std::abs(approx_integral - exact_integral) > precision) {
+        std::cout << "ERROR: " << std::setprecision(16) << "Computed integral value " << approx_integral
+                  << " does not match exact integral " << exact_integral << std::endl;
+        passed = false;
+    }
+    return passed;
+}
+
+// Test the accuracy of the exotic quadrature over some 1D functions.
+inline bool testExpMx2_sinc1_shift0() {
+    auto f = [](double x)->double {return std::exp(-x*x);};
+    bool passed = wrapSincTest1D(f, 20, 1.0, 0.0, 1.4321357541271255);
+    if (not passed) {
+        std::cout << "ERROR: Test failed in testExpMx2_sinc1_shift0()" << std::endl;
+    }
+    return passed;
+}
+inline bool testExpMx2_sinc1_shift1() {
+    auto f = [](double x)->double {return std::exp(-x*x);};
+    bool passed = wrapSincTest1D(f, 20, 1.0, 1.0, 1.4321357541271255);
+    if (not passed) {
+        std::cout << "ERROR: Test failed in testExpMx2_sinc1_shift1()" << std::endl;
+    }
+    return passed;
+}
+inline bool testExpMx2_sinc10_shift1() {
+    auto f = [](double x)->double {return std::exp(-x*x);};
+    bool passed = wrapSincTest1D(f, 20, 10.0, 1.0, 0.32099682841103033);
+    if (not passed) {
+        std::cout << "ERROR: Test failed in testExpMx2_sinc10_shift1()" << std::endl;
+    }
+    return passed;
+}
+inline bool testExpMx2_sinc100_shift1() {
+    auto f = [](double x)->double {return std::exp(-x*x);};
+    bool passed = wrapSincTest1D(f, 20, 100.0, 1.0, 0.031353648322695503);
+    if (not passed) {
+        std::cout << "ERROR: Test failed in testExpMx2_sinc100_shift1()" << std::endl;
+    }
+    return passed;
+}
+inline bool testAccuracy1D() {
+    bool passed = true;
+    passed = passed && testExpMx2_sinc1_shift0();
+    passed = passed && testExpMx2_sinc1_shift1();
+    passed = passed && testExpMx2_sinc10_shift1();
+    passed = passed && testExpMx2_sinc100_shift1();
+    return passed;
+}
+
+// Integrates the function f(x, y) * sinc(freq * x) * sinc(freq * y) over
+// [-1, 1] ^ 2  using Exotic quadrature at a specified level and shift. Then,
+// compares this integral against a given exact integral value.
+inline bool wrapSincTest2D(std::function<double(double, double)> f, int level, double freq, double shift, double exact_intgr) {
```

Comment:
One key distinction of Tasmanian (and Sparse Grid codes) is the dimension is a flexible parameter. 

Can you think of a way to pass the dimension as either a variable or a template parameter?

Reply:
Implemented by creating a more generic function called `wrapSincTest`.